src/main/java/org/truffleruby/core/string/StringSupport.java

```diff
@@ -93,6 +92,21 @@ public static int characterLength(Encoding encoding, CodeRange codeRange, byte[]
     }
 
     // Should probably use ByteLengthOfCodePointNode instead longer-term for recoverIfBroken=true cases
+    public static int characterLength(RubyEncoding encoding, TruffleString.CodeRange codeRange, byte[] bytes,
+            int byteOffset, int byteEnd, boolean recoverIfBroken) {
+        assert byteOffset >= 0 && byteOffset < byteEnd && byteEnd <= bytes.length;
+
+        if (codeRange == null) {
+            var substring = TruffleString.FromByteArrayNode.getUncached().execute(bytes, byteOffset,
+                    byteEnd - byteOffset, encoding.tencoding, false);
+
+            return TruffleString.ByteLengthOfCodePointNode.getUncached().execute(substring, 0, encoding.tencoding,
+                    TruffleString.ErrorHandling.RETURN_NEGATIVE);
+        }
```

Comment:
What's this `if` for? Notably `ErrorHandling.RETURN_NEGATIVE` is only correct if recoverIfBroken=false.

Reply:
Ah, you used `null` to replace `CR_UNKNOWN`?

Reply:
Fixed in 7d0e83e40a73deb0fff8a0055f5eca34bc1f4843